### DISCUSSION TITLE
Does the Jaeger Operator require CertManager?

### DISCUSSION THREAD

**Discussion by wiswesser** [jaegertracing/jaeger-operator#1794] [Q&A] [answered]
We recently started using Jaeger, and first installed it when version 1.27 was the current version. Since then every time a new version released, I just changed the image version for the operator. This worked fine, up to the last release (1.31.0). With this version the operator crashes, because it can not find a file in /tmp/k8s-webhook-server/serving-certs. After some digging, I realized that the new JaegerOperator Installation manifests are different. So I tried using these instead, but they cannot be deployed because the "Certificate" CRD does not exist.
So to finally get to the question:
Does the JaegerOperator now require Certmanager? For what does the Operator need Certmanager? Or can I just disable this?

I probably missed something obvious, so any help is appreciated.

Kind Regards,
Caspar Wiswesser

**Comment by frzifus** [comment, marked as answer]
Hi @wiswesser thanks for you report, i assume you are right and `cert-manager` is requiered. https://github.com/jaegertracing/jaeger-operator/issues/1761 https://github.com/jaegertracing/jaeger-operator/pull/1708

i will verify if it is so, which versions are compatible with each other and adjust the documentation accordingly if necessary.

**Reply by wiswesser** [reply]
Hey, is there any news on this yet? Or should I just try using the most recent certmanager version (This seems to be v1.7.1)?

**Reply by frzifus** [reply]
Sorry for the late response. Yes, i updated the [compatibility-matrix](https://github.com/jaegertracing/jaeger-operator#compatibility-matrix) accordingly. 
Version v1.6.1 is used for testing. But i assume everything higher then v1.0.0 should work.

**Reply by wiswesser** [reply]
Thanks for your reply, I did try simply using the newest Cert-Manager, and so far everything seems to work flawlessly :)

**Reply by dirkckau** [reply]
I got into a similar issue on it. 
May i know more about the fix on this? 

I've cert manager 1.7.2 in my cluster, however, seems the jaeger-operator is missing a mount on the cert. 
![image](https://user-images.githubusercontent.com/91056335/161468111-86ffc173-a623-445c-973e-f4a849703bc9.png)

**Reply by dirkckau** [reply]
In my case, I am ok to install the jaeger-operator after I manual created a cert and mount to the jaeger-operator. 
But, there are no placeholder on the cert in the helm chart, and I couldn't see the mutation webhook in the cluster. 

Would like to know more about the handling in your case :)

**Reply by dirkckau** [reply]
@frzifus May you help on this?

**Reply by frzifus** [reply]
I can't say much about the helm charts. The best thing to do is to create an issue on the [helm repo](https://github.com/jaegertracing/helm-charts).

Which operator version do you use? Seems they increased a few days ago to [1.32](https://github.com/jaegertracing/helm-charts/commit/55cca1253fa351de7f31e883f08ac99df42eab18).
